packages/fxa-auth-server/test/local/payments/stripe-product-and-plans-converter.js

```diff
@@ -0,0 +1,536 @@
+    let args;
+    const product1 = deepCopy({ ...product, id: 'prod_123' });
+    const productConfig1 = { stripeProductId: product1.id };
+    const productConfigId1 = '123';
+    const product2 = deepCopy({ ...product, id: 'prod_456' });
+    const productConfig2 = { stripeProductId: product2.id };
+    const productConfigId2 = '456';
+    const plan1 = deepCopy({ ...plan, id: 'plan_123' });
+    const planConfig1 = { stripePlanId: plan1.id };
+    const planConfigId1 = '123';
+    const plan2 = deepCopy({ ...plan, id: 'plan_456' });
+    const planConfig2 = { stripePlanId: plan2.id };
+    const planConfigId2 = '456';
+    const plan3 = deepCopy({ ...plan, id: 'plan_789' });
+    const planConfig3 = { stripePlanId: plan3.id };
+    const planConfigId3 = '789';
+    beforeEach(() => {
+      mockPaymentConfigManager.load = sandbox.fake.resolves();
+      mockPaymentConfigManager.startListeners = sandbox.fake.returns();
+      mockPaymentConfigManager.getDocumentIdByStripeId =
+        sandbox.fake.resolves(null);
+      converter.stripeProductToProductConfig = sandbox
+        .stub()
+        .onFirstCall()
+        .returns(productConfig1)
+        .onSecondCall()
+        .returns(productConfig2);
+      converter.stripePlanToPlanConfig = sandbox
+        .stub()
+        .onFirstCall()
+        .returns(planConfig1)
+        .onSecondCall()
+        .returns(planConfig2)
+        .onThirdCall()
+        .returns(planConfig3);
+      // For new products, this gets called twice per product
+      mockPaymentConfigManager.storeProductConfig = sandbox
+        .stub()
+        .onFirstCall()
+        .resolves(productConfigId1)
+        .onSecondCall()
+        .resolves(productConfigId1)
+        .onThirdCall()
+        .resolves(productConfigId2)
+        .onCall(3)
+        .resolves(productConfigId2);
+      mockPaymentConfigManager.storePlanConfig = sandbox
+        .stub()
+        .onFirstCall()
+        .resolves(planConfigId1)
+        .onSecondCall()
+        .resolves(planConfigId2)
+        .onThirdCall()
+        .resolves(planConfigId3);
+      mockPaymentConfigManager.stopListeners = sandbox.fake.returns();
+
+      args = {
+        productId: '',
+        isDryRun: false,
+      };
+      async function* productGenerator() {
+        yield product1;
+        yield product2;
+      }
+      async function* planGenerator1() {
+        yield plan1;
+        yield plan2;
+      }
+      async function* planGenerator2() {
+        yield plan3;
+      }
+      converter.stripeHelper.stripe = {
+        products: { list: sandbox.stub().returns(productGenerator()) },
+        plans: {
+          list: sandbox
+            .stub()
+            .onFirstCall()
+            .returns(planGenerator1())
+            .onSecondCall()
+            .returns(planGenerator2()),
+        },
+      };
+    });
+    afterEach(() => {
+      sandbox.reset();
+      Container.reset();
+    });
+    it('processes new products and plans', async () => {
```

Comment:
This test is really verbose. I haven't tried very hard, but I'm open to ways to make it more succinct.

Reply:
Discussed with Barry sync; going to blackbox these tests so I just compare the output of the method (i.e. what's in the Firestore emulator by way of PaymentConfigManager) to the expected values for `productConfig`s and `planConfig`s.